library/ssl_tls.c

```diff
@@ -11303,6 +11391,8 @@ static int ssl_write_split( mbedtls_ssl_context *ssl,
 int mbedtls_ssl_write( mbedtls_ssl_context *ssl, const unsigned char *buf, size_t len )
 {
     int ret;
+    volatile const unsigned char *buf_dup = buf;
+    volatile size_t len_dup = len;
```

Comment:
@AndrzejKurek , we can commit this implementation, but I will open task to add pointer and length to ssl ctx and check them before copy data to  ssl ctx for encryption. This way we will found out that we have a problem with pointers after our user data was already sent to outside